### PR TITLE
Expose overflow policy to Python API

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- export `OverflowPolicy` enumeration to Python
- add new `with_capacity_flush_policy` factory for `FemtoFileHandler`
- document Python overflow policy usage
- test the new policy builder

## Testing
- `make fmt`
- `make lint`
- `make typecheck`
- `make test`

------
https://chatgpt.com/codex/tasks/task_e_687043060944832296c405d606a0fd6c

## Summary by Sourcery

Expose configurable queue overflow policies to the Python API by adding an OverflowPolicy enum and a new FemtoFileHandler.with_capacity_flush_policy builder, along with documentation and tests

New Features:
- Export OverflowPolicy enum to Python
- Add with_capacity_flush_policy factory for FemtoFileHandler to set drop, block, or timeout overflow policies

Documentation:
- Document OverflowPolicy usage and the new builder in the handlers guide

Tests:
- Add tests verifying block and timeout overflow policies via the new builder

### CODE REVIEW

File: rust_extension/src/file_handler.rs [hunk 1]
```diff
@@ -220,6 +220,31 @@ impl FemtoFileHandler {
...
+            "timeout" => {
+                let ms = timeout_ms.ok_or_else(|| {
+                    PyValueError::new_err("timeout_ms required for timeout policy")
+                })?;
+                OverflowPolicy::Timeout(Duration::from_millis(ms))
+            }
+            _ => return Err(PyValueError::new_err("invalid overflow policy")),
+        };
+        Self::build_py_handler(path, capacity, Some(flush_interval), policy)
+    }
```
[the remark this answers is not in the document]
Comment (leynos) [reply rust_extension/src/file_handler.rs:246]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

[the remark this answers is not in the document]
Comment (leynos) [reply rust_extension/src/file_handler.rs:246]: @coderabbitai New GitHub issue with these details please.
